Mathlib/FieldTheory/Finite/GaloisField.lean

```diff
@@ -222,7 +223,45 @@ end GaloisField
 
 namespace FiniteField
 
-variable {K : Type*} [Field K] [Fintype K] {K' : Type*} [Field K'] [Fintype K']
+variable {K K' : Type*} [Field K] [Field K']
+
+section norm
+
+variable [Algebra K K'] [Finite K']
+
+theorem algebraMap_norm_eq_pow {x : K'} :
+    algebraMap K K' (Algebra.norm K x) = x ^ ((Nat.card K' - 1) / (Nat.card K - 1)) := by
+  have := Finite.of_injective_finite_range (algebraMap K K').injective
+  have := Fintype.ofFinite K
+  have := Fintype.ofFinite K'
+  simp_rw [← Fintype.card_eq_nat_card, Algebra.norm_eq_prod_automorphisms,
```

Comment:
I guess that the three lines above are needed to make this work; wouldn't it be more reasonable to refactor `Fintype.card_eq_nat_card` to `Finite`?

Reply:
Most of the file FieldTheory/Finite/Basic.lean (which this PR depends on) is written using `Fintype.card` rather than `Nat.card`. I've [proposed](https://leanprover.zulipchat.com/#narrow/channel/217875-Is-there-code-for-X.3F/topic/Theory.20of.20finite.20fields/near/496994060) a switch to `Nat.card` (so `Fintype` assumptions can be replaced by `Finite`), but don't have the time to implement it.
I'm not sure what's meant by "refactor `Fintype.card_eq_nat_card` to `Finite`" ...